# Calidad de sueño y estilo de vida

En este conjunto de datos sacado de [Kaggle](https://www.kaggle.com/datasets/uom190346a/sleep-health-and-lifestyle-dataset) cuenta con datos de la calidad de sueño de distintas personas, las columnas que se encuentran son las siguientes:

- Person ID: Un identificador para cada individuo
- Gener: El género de la persona (Hombre / Mujer)
- Age: La edad de las personas en años
- Occupation: - La ocupación o profesión de la persona
- Sleep Duration (hours) - El número de horas que la persona duerme por día
- Quality Sleep (scale: 1-10) - Una calificación subjetiva de la calidad del sueño
- Physical Activity Level (minutes/day) - El número de minutos que la persona hace actividad física diariamente
- Stress Level (scale: 1-10) - Una calificacoón subjetiva del nivel de estres experimentado por la persona
- BMI Category - La categoría de Índice de Masa Corporal
- Blood Pressure (systolic/diastolic) - La presión arterial de la persona, indicado como presión sistólica sobre presión diastólica
- Heart Rate (bpm) - La frecuencia cardíaca en reposo en latidos por minuto
- Dialy Steps - El número de pasos que la persona hace al día
- Sleep Disorder - La presencia o ausensica de transtornos del sueño en la persona.

## Preguntas que buscamos responder con este conjunto

Las preguntas que buscaremos responder con este conjunto de datos son las siguientes:

- ¿Quiénes tienen peor calidad del sueño, hombres o mujeres?

- ¿Existe una relación entre la calidad del sueño de las personas y su profesión?

- ¿La actividad física afecta al sueño?

- ¿Qué profesión presenta la peor calidad de sueño?

- ¿Cuál profesión tienen a las personas con el mayor nivel de estrés y el mayor índice de masa corporal?

- ¿En qué rango de edades se encuentran la mayor cantidad de trastornos del sueño?

- La cantidad de pasos al día, ¿afecta la calidad del sueño? ¿al índice de masa corporal?

Con estas preguntas comenzaremos a hacer nuestro análisis buscando extraer la mayor cantidad de información que se pueda del conjunto, es importante señalar que en este proyecto solamente haremos análisis de datos utilizando la librería **Pandas**.



In [ ]:
import pandas as pd

In [ ]:
# Cargamos el conjunto de datos
data = pd.read_csv(
      '/content/sample_data/sleep health.csv',
      index_col="Person ID"
    )
data.head()

Gender  Age            Occupation  Sleep Duration  Quality of Sleep  \
Person ID                                                                       
1           Male   27     Software Engineer             6.1                 6   
2           Male   28                Doctor             6.2                 6   
3           Male   28                Doctor             6.2                 6   
4           Male   28  Sales Representative             5.9                 4   
5           Male   28  Sales Representative             5.9                 4   

           Physical Activity Level  Stress Level BMI Category Blood Pressure  \
Person ID                                                                      
1                               42             6   Overweight         126/83   
2                               60             8       Normal         125/80   
3                               60             8       Normal         125/80   
4                               30             8        Obese         140/90   
5                               30             8        Obese         140/90   

           Heart Rate  Daily Steps Sleep Disorder  
Person ID                                          
1                  77         4200           None  
2                  75        10000           None  
3                  75        10000           None  
4                  85         3000    Sleep Apnea  
5                  85         3000    Sleep Apnea

In [ ]:
# Información general del conjunto
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 374 entries, 1 to 374
Data columns (total 12 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Gender                   374 non-null    object 
 1   Age                      374 non-null    int64  
 2   Occupation               374 non-null    object 
 3   Sleep Duration           374 non-null    float64
 4   Quality of Sleep         374 non-null    int64  
 5   Physical Activity Level  374 non-null    int64  
 6   Stress Level             374 non-null    int64  
 7   BMI Category             374 non-null    object 
 8   Blood Pressure           374 non-null    object 
 9   Heart Rate               374 non-null    int64  
 10  Daily Steps              374 non-null    int64  
 11  Sleep Disorder           374 non-null    object 
dtypes: float64(1), int64(6), object(5)
memory usage: 38.0+ KB


In [ ]:
# Descripción general de las variables numericas del conjunto
data.describe()

Age  Sleep Duration  Quality of Sleep  Physical Activity Level  \
count  374.000000      374.000000        374.000000               374.000000   
mean    42.184492        7.132086          7.312834                59.171123   
std      8.673133        0.795657          1.196956                20.830804   
min     27.000000        5.800000          4.000000                30.000000   
25%     35.250000        6.400000          6.000000                45.000000   
50%     43.000000        7.200000          7.000000                60.000000   
75%     50.000000        7.800000          8.000000                75.000000   
max     59.000000        8.500000          9.000000                90.000000   

       Stress Level  Heart Rate   Daily Steps  
count    374.000000  374.000000    374.000000  
mean       5.385027   70.165775   6816.844920  
std        1.774526    4.135676   1617.915679  
min        3.000000   65.000000   3000.000000  
25%        4.000000   68.000000   5600.000000  
50%        5.000000   70.000000   7000.000000  
75%        7.000000   72.000000   8000.000000  
max        8.000000   86.000000  10000.000000

In [ ]:
data.corr().columns

<ipython-input-7-ea46686d89d7>:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  data.corr().columns


Index(['Age', 'Sleep Duration', 'Quality of Sleep', 'Physical Activity Level',
       'Stress Level', 'Heart Rate', 'Daily Steps'],
      dtype='object')

### Análisis de las edades

In [ ]:
# Cantidad de hombres y mujeres en el conjunto
data['Gender'].value_counts()

Male      189
Female    185
Name: Gender, dtype: int64

In [ ]:
# Promedio de la edad
data['Age'].mean()

42.18449197860963

In [ ]:
# Desviación estándar del conjunto
data['Age'].std()

8.673133465547243

In [ ]:
# Valor mínimo de las edades
data['Age'].min()

27

In [ ]:
# Valor máximo de las edades
data['Age'].max()

59

In [ ]:
# Creamos una columna nueva para ordenar por adulto y adulto viejo
data.loc[data['Age']<=40, 'Age_group'] = 'Adult'
data.loc[data['Age']>=41, 'Age_group'] = 'Older Adult'

In [ ]:
# Cantidad de adultos y adultos viejos del conjunto
data['Age_group'].value_counts()

Older Adult    209
Adult          165
Name: Age_group, dtype: int64

In [ ]:
# Promedio y desviación estándar de las edades divididas entre hombres y mujeres
data.groupby('Gender')['Age'].agg(['mean', 'std'])

mean       std
Gender                     
Female  47.405405  8.093407
Male    37.074074  5.662006

## Análisis de las ocupaciones

In [ ]:
# Cantidad de hombres y mujeres
data.groupby(['Occupation'])['Gender'].value_counts()

Occupation            Gender
Accountant            Female    36
                      Male       1
Doctor                Male      69
                      Female     2
Engineer              Female    32
                      Male      31
Lawyer                Male      45
                      Female     2
Manager               Female     1
Nurse                 Female    73
Sales Representative  Male       2
Salesperson           Male      32
Scientist             Female     4
Software Engineer     Male       4
Teacher               Female    35
                      Male       5
Name: Gender, dtype: int64

In [ ]:
# Duración mínima y máxima del sueño de las personas
print(data['Sleep Duration'].min())
print(data['Sleep Duration'].max())

5.8
8.5


In [ ]:
# Duración del sueño entre hombres y mujeres
data.groupby(['Gender'])['Sleep Duration'].mean()

Gender
Female    7.229730
Male      7.036508
Name: Sleep Duration, dtype: float64

In [ ]:
data.groupby('Gender')['Quality of Sleep'].mean()

Gender
Female    7.664865
Male      6.968254
Name: Quality of Sleep, dtype: float64

In [ ]:
# Promedio de la calidad y duración del sueño
occ_qua=data.groupby('Occupation')[['Quality of Sleep', 'Sleep Duration']].mean()
occ_qua.sort_values(by='Quality of Sleep')

Quality of Sleep  Sleep Duration
Occupation                                            
Sales Representative          4.000000        5.900000
Scientist                     5.000000        6.000000
Salesperson                   6.000000        6.403125
Software Engineer             6.500000        6.750000
Doctor                        6.647887        6.970423
Teacher                       6.975000        6.690000
Manager                       7.000000        6.900000
Nurse                         7.369863        7.063014
Accountant                    7.891892        7.113514
Lawyer                        7.893617        7.410638
Engineer                      8.412698        7.987302

In [ ]:
occ_bmi = data.groupby('Occupation')['BMI Category'].value_counts()

occ_bmi.sort_values()

Occupation            BMI Category 
Lawyer                Normal Weight     1
Software Engineer     Overweight        1
                      Obese             1
Manager               Overweight        1
Teacher               Obese             1
Doctor                Normal Weight     2
Lawyer                Obese             2
                      Overweight        2
Software Engineer     Normal Weight     2
Sales Representative  Obese             2
Engineer              Overweight        3
                      Normal Weight     4
Doctor                Obese             4
Scientist             Overweight        4
Accountant            Normal Weight     5
                      Overweight        6
Teacher               Normal            6
Nurse                 Normal Weight     7
Accountant            Normal           26
Salesperson           Overweight       32
Teacher               Overweight       33
Lawyer                Normal           42
Engineer              Normal           5

In [ ]:
occ_steps = data.groupby(['Occupation'])['Daily Steps', 'Physical Activity Level', 'Sleep Duration', 'Quality of Sleep'].mean()
occ_steps.sort_values(by='Daily Steps')

<ipython-input-32-a128c6d93949>:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  occ_steps = data.groupby(['Occupation'])['Daily Steps', 'Physical Activity Level', 'Sleep Duration', 'Quality of Sleep'].mean()


Daily Steps  Physical Activity Level  Sleep Duration  \
Occupation                                                                   
Sales Representative  3000.000000                30.000000        5.900000   
Scientist             5350.000000                41.000000        6.000000   
Manager               5500.000000                55.000000        6.900000   
Software Engineer     5800.000000                48.000000        6.750000   
Teacher               5957.500000                45.625000        6.690000   
Engineer              5980.952381                51.857143        7.987302   
Salesperson           6000.000000                45.000000        6.403125   
Doctor                6808.450704                55.352113        6.970423   
Accountant            6881.081081                58.108108        7.113514   
Lawyer                7661.702128                70.425532        7.410638   
Nurse                 8057.534247                78.589041        7.063014   

                      Quality of Sleep  
Occupation                              
Sales Representative          4.000000  
Scientist                     5.000000  
Manager                       7.000000  
Software Engineer             6.500000  
Teacher                       6.975000  
Engineer                      8.412698  
Salesperson                   6.000000  
Doctor                        6.647887  
Accountant                    7.891892  
Lawyer                        7.893617  
Nurse                         7.369863

# Reporte

# Introducción

En este proyecto tomamos el conjunto de datos [Sleep Health and Lifestyle Dataset | Kaggle](https://www.kaggle.com/datasets/uom190346a/sleep-health-and-lifestyle-dataset) en donde se encuentran los datos de distintas personas como su profesión, su actividad física y la calidad de sueño. Tomaremos este conjunto de datos para conocer que cosas son las que afectan al sueño como por ejemplo la cantidad de actividad física, la profesión y la presión a la que cada uno de ellos se somete.

Las preguntas que buscaremos responder con los datos de este conjunto son las siguientes:

- ¿Quiénes tienen peor calidad del sueño, hombres o mujeres?

- ¿Existe una relación entre la calidad del sueño de las personas y su profesión?

- ¿La actividad física afecta al sueño?

- ¿Qué profesión presenta la peor calidad de sueño?

- ¿Cuál profesión tienen a las personas con el mayor nivel de estrés y el mayor índice de masa corporal?

- ¿En qué rango de edades se encuentran la mayor cantidad de trastornos del sueño?

- La cantidad de pasos al día, ¿afecta la calidad del sueño? ¿al índice de masa corporal?

Estas preguntas nos servirán como punto de partida en nuestro análisis buscando extraer la mayor cantidad de información que se pueda del conjunto, es importante señalar que para el análisis de este conjunto de datos solamente haremos uso de la librería **Pandas** por lo que en este análisis no encontraremos gráficos.

## Análisis del conjunto

### Análisis General

Sabemos que en el conjunto de datos contamos con *374* filas y *13* columnas de datos, de las cuales sabemos que en general los datos se dividen de la siguiente forma:

|Género|Cantidad|
|---|---|
|Hombres|189|
|Mujeres|185|

Esto nos da un vistazo general del conjunto y podemos decir que el conjunto se encuentran en proporción, respecto a la cantidad de hombre y mujeres.

## Clasificación dada por edad

En el conjunto podemos encontrar las edades de las personas, con respecto a las edades podemos encontrar lo siguiente:

- La edad más baja es de *27* años

- La edad más alta es de *59* años

Lo que podemos decir que el conjunto de datos toma en cuenta a personas que ya se encuentran en la edad laboral iniciando en el trabajo o con pocos años en el mismo, o de personas cercanas a jubilarse.

La edad promedio del conjunto de datos es de *42* años y la desviación estándar es de *8.67*, lo que quiere decir que en este conjunto podemos encontrar datos muy dispersos lo que puede deberse a los pocos datos recabados.  

Para apoyarnos en este análisis se agregó una nueva columna al conjunto de datos a la que se le llamó `Age_group` en la cuál se clasifico la edad de las personas de la siguiente manera:  

- Edad menor o igual a 40: `Adulto`

- Edad mayor a 40: `Adulto viejo`  

Tomando en cuenta este ordenamiento de los datos tenemos lo siguiente:

|Clasificación|Cantidad|
|---|---|
|Adulo Viejo|209|
|Adulto|165|


Finalmente el promedio de edades por cada género queda de la siguiente forma:  

|Gender|mean|std|
|---|---|---|
|Female|47\.40|8\.10|
|Male|37\.07|5\.66|

Podemos ver que existe una diferencia de `10` años en el promedio de edades de ambos géneros, y también notamos un diferencia grande en la desviación estándar, siendo ésta de `3` años.


## Ocupaciones

|Ocupación |Género|Cantidad|
|---|---|---|
|Accountant|Female|36|
||Male|1|
|Doctor|Male|69|
||Female|2|
|Engineer|Female|32|
||Male|31|
|Lawyer|Male|45|
|| Female|2|
|Manager|Female|1|
|Nurse|Female|73|
|Sales Representative|Male|2|
|Salesperson|Male|32|
|Scientist|Female|4|
|Software Engineer|Male|4|
|Teacher|Female|35|
||Male|5|

En esta gráfica notamos la siguiente:
- Las ocupaciones en donde hay más presencia de mujeres son:
	- Contador
	- Ingeniería
	- Gerente
	- Enfermería
	- Ciencia
	- Maestro
- Las ocupaciones en donde los hombres tienen más presencia:
	- Doctor
	- Representante de ventas
	- Ventas
	- Abogado
	- Ingeniero de Software

Es importante mencionar que solamente en *Ingeniería* la diferencia es de solamente una persona.

### Calidad del Sueño

El promedio de la duración del sueño entre hombres y mujeres es la siguiente:

|Genero|Promedio del sueño|Calidad del sueño|
|---|---|---|
|Female|7.23|7.66|
|Male|7.04|6.97|

Podemos ver que en general, las mujeres tienen un mejor sueño que los hombres no solamente en cantidad sino que en calidad, para profundizar más, obtendremos los mismos datos pero ahora en cada una de las profesiones del conjunto.

|Occupation|Quality of Sleep|Sleep Duration|
|---|---|---|
|Sales Representative|4\.0|5\.9|
|Scientist|5\.0|6\.0|
|Salesperson|6\.0|6\.40|
|Software Engineer|6\.5|6\.75|
|Doctor|6\.65|6\.97|
|Teacher|6\.975|6\.69|
|Manager|7\.0|6\.9|
|Nurse|7\.37|7\.06|
|Accountant|7\.89|7\.11|
|Lawyer|7\.89|7\.41|
|Engineer|8\.41|7\.99|


En estas tablas podemos ver que las peores ocupaciones que tienen la peor calidad y cantidad de sueño son:
- Representante de Ventas
- Científico
- Ventas

Y aquellas que tienen mejor calidad y tiempo de sueño son:
- Ingeniero
- Abogado
- Contador

En las profesiones donde peor calidad y cantidad de sueño hay se encuentran más hombres que mujeres.

### Problemas de salud

Finalmente buscaremos que tan buena salud tienen cada una de las personas en las distintas profesiones, con este dato podremos sacar nuestras conclusiones y podremos responder las preguntas que nos hicimos en un inicio al comenzar con nuestro análisis.

| Occupation           | BMI Category    | Count |
|----------------------|-----------------|-------|
| Lawyer               | Normal Weight   | 1     |
| Software Engineer    | Overweight      | 1     |
|                      | Obese           | 1     |
| Manager              | Overweight      | 1     |
| Teacher              | Obese           | 1     |
| Doctor               | Normal Weight   | 2     |
| Lawyer               | Obese           | 2     |
|                      | Overweight      | 2     |
| Software Engineer    | Normal Weight   | 2     |
| Sales Representative | Obese           | 2     |
| Engineer             | Overweight      | 3     |
|                      | Normal Weight   | 4     |
| Doctor               | Obese           | 4     |
| Scientist            | Overweight      | 4     |
| Accountant           | Normal Weight   | 5     |
|                      | Overweight      | 6     |
| Teacher              | Normal          | 6     |
| Nurse                | Normal Weight   | 7     |
| Accountant           | Normal          | 26    |
| Salesperson          | Overweight      | 32    |
| Teacher              | Overweight      | 33    |
| Lawyer               | Normal          | 42    |
| Engineer             | Normal          | 56    |
| Doctor               | Normal          | 65    |
| Nurse                | Overweight      | 66    |

Otro punto importante a tener en cuenta es saber si la cantidad de pasos al día afecta a la salud y a la calidad del sueño, para conocer esto tendremos el promedio de los pasos diarios, el nivel de actividad física, duración del sueño y la calidad del sueño en cada una de las profesiones.

|Occupation|Daily Steps|Physical Activity Level|Sleep Duration|Quality of Sleep|
|---|---|---|---|---|
|Sales Representative|3000\.0|30\.0|5\.9|4\.0|
|Scientist|5350\.0|41\.0|6\.0|5\.0|
|Manager|5500\.0|55\.0|6\.9|7\.0|
|Software Engineer|5800\.0|48\.0|6\.75|6\.5|
|Teacher|5957\.5|45\.62|6\.69|6\.97|
|Engineer|5980\.95|51\.86|7\.99|8\.41|
|Salesperson|6000\.0|45\.0|6\.40|6\.0|
|Doctor|6808\.45|55\.35|6\.97|6\.65|
|Accountant|6881\.08|58\.11|7\.11|7\.89|
|Lawyer|7661\.70|70\.42|7\.41|7\.89|
|Nurse|8057\.53|78\.59|7\.06|7\.37|

<div style="page-break-after: always; visibility: hidden">
\pagebreak
</div>

# Conclusiones

Este conjunto de datos nos sirve para darnos un vistazo general de cada una de las profesiones y como éstas afecta a la salud de las personas. Digo general porque un problema de este conjunto es la cantidad de datos que tiene, ya que al contar solamente con *307* filas son muy pocas para todo el margen de edad que se tomó. Eso lo podemos notar al obtener el promedio de las edades y la desviación estándar por lo que, al terminar con nuestro análisis, podemos decir que hacen falta más datos o en su caso reducir el margen de edades de los encuestados.

Pasando al sueño de cada una de las personas tenemos lo siguiente:

Las mujeres en términos generales tienen una mejor calidad y cantidad de sueño respecto a los hombres ya que, las mujeres pueden tener un mejor control de estrés o porque las mujeres a las que se les hizo la encuesta parecen tener profesiones en las que se habla con menos personas que las profesiones de los hombres.

Esto lo podemos acompañar si tomamos solamente las primeras cinco profesiones que presentan la mejor calidad y cantidad de sueño:

- Ingeniería
- Abogado
- Contador
- Enfermería
- Gerente

De estas profesiones en tres se encuentran más mujeres (Contador, Enfermería y Gerente) en una hay más hombres (Abogado) y en Ingeniería se encuentra la diferencia de una persona entre ambos.

Muchas de estas profesiones también se repiten cuando revisamos la cantidad de pasos al día, podemos ver que las cinco profesiones donde más pasos al día se dan son:

- Enfermería
- Abogado
- Contador
- Doctor
- Vendedor

En la mayoría de estas personas podemos ver que caminar forma una parte fundamental de ellas por lo que no es de extrañar que éstas presenten un promedio alto de pasos al día y esto también se ve en la salud del sueño, ya que si vemos la tabla de los pasos al día podemos ver que en ella existe una relación que entre más pasos se den mejor va a ser el sueño.

Aunque esta misma relación no se nota cuando hablamos del índice de masa corporal, porque de la lista anterior podemos ver que en las siguientes se encuentran muchas personas con sobrepeso:

- Enfermería
- Vendedor

Finalmente, las preguntas hechas alrededor de la edad de los encuestados no se respondieron ya que se encontró que el rango de edades es muy alto y se cuenta con muy pocos datos por lo que las respuestas que se puedan encontrar en este ámbito no pueden llegar a ser relevantes con la realidad.